ui/src/wallet.ts

```diff
@@ -0,0 +1,132 @@
+import WalletConnect from "@walletconnect/client";
+import { convertUtf8ToHex } from "@walletconnect/utils";
+import QRCodeModal from "@walletconnect/qrcode-modal";
+import WalletConnectWeb3Provider from "@walletconnect/web3-provider";
+import Web3 from "web3";
+import { writable, Readable } from "svelte/store";
+
+type State =
+  | { status: "NOT_CONNECTED" }
+  | { status: "CONNECTED"; connected: Connected };
+
+interface Connected {
+  chainId: number;
+  account: {
+    address: string | null;
+    balance: string | null;
+  };
+}
+
+interface Wallet extends Readable<State> {
+  connect(): Promise<void>;
+  disconnect(): Promise<void>;
+  testSign(): Promise<void>;
+}
+
+export function makeWallet(): Wallet {
+  const stateStore = writable<State>({ status: "NOT_CONNECTED" });
+  const connector = new WalletConnect({
+    bridge: "https://bridge.walletconnect.org",
```

Comment:
I see doing so being useful if the config is environment-dependent but simply moving these hardcoded values to a different file while they are used in a single place doesn't seem to add much value. Could you explain what benefits you see?

Reply:
I think it would be easier to find if all the configuration knobs have meaningful names and in one place, so I don't have to hunt around unfamiliar code.

But yeah, as this is just an experiment, perhaps it's too early for that.

Reply:
Will there ever be a different walletconnect endpoint to be used?

Reply:
don't think so. that's also the value that walletconnect uses in their sample project and haven't seen another alternative.